src/libs/actions/PersistedRequests.js

```diff
@@ -0,0 +1,58 @@
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+import lodashUnionWith from 'lodash/unionWith';
+import ONYXKEYS from '../../ONYXKEYS';
+import RetryCounter from '../Network/RetryCounter';
+
+const persistedRequestsRetryCount = new RetryCounter();
```

Comment:
Personally, I think this `RetryCounter` should live in `src/libs/Network/PersistedRequestQueue.js` rather than here. That way `src/libs/actions/PersistedRequests.js` just deals with manipulating Onyx data and the ephemeral `RetryCounter` data (stored only in memory, not on disk), is managed in the `PersistedRequestQueue` directly.

Reply:
I'm gonna disagree here even though I felt a similar impulse at one point. I might also have a slightly different interpretation about whether an action file should only deal with Onyx data manipulation (I think it's fine to have it do other things like the side effects for the retry counter). More importantly, if we move this then we'll need to import it into the actions file and that will create a circular dependency.